lte/gateway/release/promote.py

```diff
@@ -0,0 +1,135 @@
+           else:
+              print(f"GET call failed with status Code={response.status_code}. Response={response.json()}")
+              return -1
+        except requests.exceptions.HTTPError as error:
+              print(f"HTTPError Occurred. Error=")
+              print(error)
+              return -1
+    elif method == 'POST':
+        try:
+           response = requests.post(url + '/' + sourceRepo + '/v2/promote', headers=reqHeaders, json=payload, auth=(JFROG_USER, JFROG_API_KEY))
+           if(response.status_code == 200):
+              print(f"SUCCESS promoting {imgName}:{imgVersion} from {sourceRepo} to {targetRepo}.")
+              print(f"Response Code={response.status_code}")
+              print(f"Response Body={response.content}")
+           else:
+              print(f"Response Code={response.status_code}")
+              print(f"Response Body={response.json()}")
+              return -1
+        except requests.exceptions.HTTPError as error:
+              print("HTTPError Occurred. Error=")
+              print(error)
+              return -1
+    else:
+        print(f"INCORRECT API method {method} invoked. Please use GET or POST.")
+        return -1
+
+    return 0
+
+def validate(module):
+    print(f"Validating {imgName}:{imgVersion} in repop:{sourceRepo}")
+    status = callJFROGAPI(method='GET')
+    return status
+
+
+def promote(module):
+    print(f"Promoting {imgName}:{imgVersion} from {sourceRepo} to {targetRepo}")
+    status = callJFROGAPI(method='POST', payload={ "targetRepo": targetRepo, "dockerRepository": imgName, "tag": imgVersion, "copy": "true" })
+    return status
+
+def _get_jfrog_apikey():
+    try:
+        with open(os.path.expanduser(JFROG_APIKEY_PATH), 'r') as keyfile:
+            return keyfile.read().rstrip()
+    except IOError as e:
+        print(e)
+        abort("Please add the Jfrog API key for user %s to %s and re-run"
+              % (JFROG_USER, JFROG_APIKEY_PATH))
+
+JFROG_API_KEY = _get_jfrog_apikey()
+
+if __name__ == "__main__":
+    modules = [ 'docker', 'feg', 'cwf', 'orc8r', 'all' ]
+    argCount = len(sys.argv)
+
+    if argCount != 4:
+       print(f"Invalid arguments. Usage: {sys.argv[0]} {modules} imageName version. Ex: promote.py feg gateway_go 1a4b2b30")
+       abort("Exiting...")
+
+    if sys.argv[1] not in modules:
+       print(f"Module {sys.argv[1]} not found in allowed list {modules}")
+       print(f"Usage: {sys.argv[0]} {modules} imageName version")
+       abort("Exiting...")
+
+    module = sys.argv[1]
+    imgName = sys.argv[2]
+    imgVersion = sys.argv[3]
+    sourceRepo = module + "-test"
+    targetRepo = module + "-prod"
+
+
+    status = validate(module)
+    if status == 0:
+       print(f"Validation of {imgName}:{imgVersion} in repop:{sourceRepo} is successful.")
+    else:
+       print(f"Validation of {imgName}:{imgVersion} in repop:{sourceRepo} FAILED. CANNOT Promote. Quitting...")
+       quit()
+
+    status = promote(module)
```

Comment:
There is some logic missing. 
what if I want to promote all images? 
what if I only want the agw one? 

Let's do fab function:
1 for docker 
1 for agw 
1 for helm